src/main/java/org/folio/eusage/reports/api/EusageReportsApi.java

```diff
@@ -245,43 +245,33 @@ private void endStream(RowStream<Row> stream, RoutingContext ctx,
   }
 
   Future<UUID> upsertTeEntry(TenantPgPool pool, SqlConnection con, RoutingContext ctx,
-                             String counterReportTitle, String match) {
+                             String counterReportTitle, String printIssn, String onlineIssn) {
     return con.preparedQuery("SELECT id FROM " + titleEntriesTable(pool)
         + " WHERE counterReportTitle = $1")
         .execute(Tuple.of(counterReportTitle))
         .compose(res1 -> {
           if (res1.iterator().hasNext()) {
             return Future.succeededFuture(res1.iterator().next().getUUID(0));
           }
+          final String match = onlineIssn;
           return ermLookup(ctx, match).compose(erm -> {
-            Future<Void> future;
-            if (erm == null) {
-              future = con.preparedQuery("INSERT INTO " + titleEntriesTable(pool)
-                  + "(id, counterReportTitle, matchCriteria,"
-                  + " kbManualMatch)"
-                  + " VALUES ($1, $2, $3, $4)"
-                  + " ON CONFLICT (counterReportTitle) DO NOTHING")
-                  .execute(Tuple.tuple(List.of(UUID.randomUUID(), counterReportTitle, match,
-                      false))).mapEmpty();
-            } else {
-              UUID kbTitleId = erm.getUUID(0);
-              String kbTitleName = erm.getString(1);
-              future = con.preparedQuery("INSERT INTO " + titleEntriesTable(pool)
-                  + "(id, counterReportTitle, matchCriteria,"
-                  + " kbTitleName, kbTitleId,"
-                  + " kbManualMatch)"
-                  + " VALUES ($1, $2, $3, $4, $5, $6)"
-                  + " ON CONFLICT (counterReportTitle) DO NOTHING")
-                  .execute(Tuple.tuple(List.of(UUID.randomUUID(), counterReportTitle, match,
-                      kbTitleName, kbTitleId,
-                      false))).mapEmpty();
-            }
-            return future.compose(x ->
-                con.preparedQuery("SELECT id FROM " + titleEntriesTable(pool)
-                    + " WHERE counterReportTitle = $1")
-                    .execute(Tuple.of(counterReportTitle))
-                    .map(res2 -> res2.iterator().next().getUUID(0))
-            );
+            UUID kbTitleId = erm != null ? erm.getUUID(0) : null;
+            String kbTitleName = erm != null ? erm.getString(1) : null;
+            return con.preparedQuery("INSERT INTO " + titleEntriesTable(pool)
+                + "(id, counterReportTitle, matchCriteria,"
```

Comment:
@adamdickmeiss In the design doc we have moved the counterReportTitle to the TITLE_DATA table.

Reply:
Yes. That seems very strange by now.

Reply:
I will change this in a future PR.